Problem2/tc_largest_two_numbers.rb

```diff
@@ -0,0 +1,60 @@
+# File:  tc_largest_two_numbers.rb
+
+require_relative "largest_two_numbers.rb"
+require "test/unit"
+
+class TestLargesTwoNumbers < Test::Unit::TestCase
+	
+	def test_empty
+		
+		largest_two_numbers = LargestTwoNumbers.new
+		numbers = Array[]	
+		
+		assert_raise RuntimeError do
+			returned_result = largest_two_numbers.get_largest_two_numbers(numbers)	
+		end		
+	end
+	
+		
+	def test_below_minimun
+		
+		largest_two_numbers = LargestTwoNumbers.new
+		numbers = Array[5]
+		
+		assert_raise RuntimeError do
+			returned_result = largest_two_numbers.get_largest_two_numbers(numbers)	
+		end		
+	end
+	
+	def test_just_above_minimum
+		
+		largest_two_numbers = LargestTwoNumbers.new
+		numbers = Array[0,5]
+		
+		returned_result = largest_two_numbers.get_largest_two_numbers(numbers)		
+		
+		assert_equal(Array[5,0], returned_result)
+	end
+
+	
+	def test_normal_array
+		
+		largest_two_numbers = LargestTwoNumbers.new
+		numbers = Array[5,7,9,10,1000,9999,55,12,-50,15000]
+		
+		returned_result = largest_two_numbers.get_largest_two_numbers(numbers)		
+		
+		assert_equal(Array[15000,9999], returned_result)
+	end
+	
+	def test_with_duplicates
+		
+		largest_two_numbers = LargestTwoNumbers.new
+		numbers = Array[5.9,10,12,10,78,78]
+		
+		returned_result = largest_two_numbers.get_largest_two_numbers(numbers)		
+		
+		assert_equal(Array[78,78], returned_result)
```

Comment:
The text/requirement for this problem is a bit ambiguous and might be considered incomplete (which happens often in the real world). It requires that the developer thinks a bit about the way his code will be used. In my opinion if a client requested of me to write a piece of code which finds the 2 largest numbers in an array, that piece of code would return "the 2 largest numbers in the array". "78" is the same number as "78". 10, 12 and so on are different numbers. If I'm allowed to be a bit evil, [here](https://en.wikipedia.org/wiki/Number) is the definition of a number.